lambda/service/handler/get_pennsieve_accounts_handler.go

```diff
@@ -0,0 +1,71 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/pennsieve/account-service/service/models"
+	"github.com/pennsieve/account-service/service/utils"
+)
+
+const (
+	AWS = "aws"
+)
+
+func GetPennsieveAccountsHandler(request events.APIGatewayV2HTTPRequest) (events.APIGatewayV2HTTPResponse, error) {
+	handlerName := "GetPennsieveAccountsHandler"
+	accountType := utils.ExtractParam(request.RawPath)
+	log.Println("request.RequestContext.AccountID", request.RequestContext.AccountID)
+	ctx := context.Background()
+
+	switch strings.ToLower(accountType) {
+	case AWS:
+		cfg, err := config.LoadDefaultConfig(ctx)
+		if err != nil {
+			log.Println(err.Error())
+			return events.APIGatewayV2HTTPResponse{
+				StatusCode: 500,
+				Body:       handlerName,
+			}, ErrConfig
+		}
+
+		client := sts.NewFromConfig(cfg)
+		input := &sts.GetCallerIdentityInput{}
+
+		req, err := client.GetCallerIdentity(ctx, input)
+		if err != nil {
+			log.Println(err.Error())
+			return events.APIGatewayV2HTTPResponse{
+				StatusCode: 500,
+				Body:       handlerName,
+			}, ErrSTS
+		}
+		accountId := *req.Account
+		m, err := json.Marshal(models.PennsieveAccount{
+			AccountID: accountId,
+			Type:      AWS,
+		})
+		if err != nil {
+			log.Println(err.Error())
+			return events.APIGatewayV2HTTPResponse{
+				StatusCode: 500,
+				Body:       handlerName,
+			}, ErrMarshaling
+		}
+		response := events.APIGatewayV2HTTPResponse{
+			StatusCode: 200,
+			Body:       string(m),
+		}
+		return response, nil
+	default:
+		return events.APIGatewayV2HTTPResponse{
+			StatusCode: 422,
+			Body:       handlerName,
+		}, ErrUnsupportedAccountType
```

Comment:
One thing I learned in the rehydration-service, is that if you return an error from this main handler, the AWS API Gateway code ignores the StatusCode and Body in the APIGatewayV2HTTPResponse and just returns a 500 "internal service error" to the caller. So if we want the user to see this 422 status code and Body, the error part needs to be nil.

And the same for the errors above, but maybe less important since the desired status code is 500 anyway. But the body would be lost.

Reply:
thanks @jfromme - will update

Reply:
added, thanks @jfromme